src/lib/packet.h

```diff
@@ -377,6 +377,7 @@ typedef enum {
                                * IETF-S/MIME) */
     PGP_PKA_EDDSA = 22,       /* EdDSA from draft-ietf-openpgp-rfc4880bis */
 
+    PGP_PKA_SM2_ENCRYPT = 98, /* SM2 encryption */
```

Comment:
Yeah I wasn't so sure about this point either. In general, best practice is to not use the same key for both encryption + signing because sometimes there are strange cross-effects that can occur. But it's not really clear from the docs that I have seen what the convention is for SM2. One hint however is that there are distinct OIDs assigned for SM2 encryption, signature, and key exchange - that to me suggests that each keypair is only supposed to be used for one of the operations.